File: src/main/java/wraith/fabricaeexnihilo/modules/witchwater/WitchWaterStatusEffect.java
package wraith.fabricaeexnihilo.modules.witchwater;

import net.minecraft.entity.effect.StatusEffect;
import net.minecraft.entity.effect.StatusEffectCategory;
import net.minecraft.entity.effect.StatusEffectInstance;
import wraith.fabricaeexnihilo.util.Color;

public class WitchWaterStatusEffect extends StatusEffect {

    public WitchWaterStatusEffect() {
        super(StatusEffectCategory.NEUTRAL, Color.DARK_PURPLE.toInt());
    }

    public StatusEffectInstance getInstance() {
        return new StatusEffectInstance(this, 72000, 1, false, false, false);
    }

}

File: src/main/kotlin/exnihilofabrico/modules/witchwater/WitchWaterStatusEffect.kt
package exnihilofabrico.modules.witchwater

import exnihilofabrico.util.Color
import net.minecraft.entity.effect.StatusEffect
import net.minecraft.entity.effect.StatusEffectInstance
import net.minecraft.entity.effect.StatusEffectType

object WitchWaterStatusEffect: StatusEffect(StatusEffectType.NEUTRAL, Color.DARK_PURPLE.toInt()) {

    fun getInstance(): StatusEffectInstance {
        val instance = StatusEffectInstance(this, 72000, 1, false, false, false)
        instance.isPermanent = true

        return instance
    }

}